src/charm.py

```diff
@@ -389,26 +383,40 @@ def _config_changed(self, event: HookEvent) -> None:
                 else "",
             )
 
-        # First execute the setup script in 2 conditions:
+        # Execute the commands on 2 conditions:
         # - First run (when no services are planned in pebble)
         # - Change in important S3 parameter (comparing value with envVars in pebble plan)
-        if (
-            self._is_config_valid()
-            and self.model.unit.is_leader()
-            and self._should_run_setup(current_plan, previous_s3_info)
-        ):
-            self.model.unit.status = MaintenanceStatus("Compiling assets")
-            script = f"{SCRIPT_PATH}/pod_setup.sh"
-            process = container.exec(
-                [script],
-                environment=self._create_discourse_environment_settings(),
-                working_dir="/srv/discourse/app",
-            )
+        if self._is_config_valid() and self.model.unit.is_leader() and not current_plan.services:
+            env_settings = self._create_discourse_environment_settings()
             try:
-                stdout, _ = process.wait_output()
-                logger.debug("%s stdout: %s", script, stdout)
+                self.model.unit.status = MaintenanceStatus("Running migrations")
```

Comment:
Should we include an upgrade test for this?

Reply:
I closed this PR for now but I'm replying anyway.
The migrations are already run on installation. The was Rails handles migrations make unnecessary to run an upgrade test. All changes are applied iteratively in a deterministic order (the migration file names are prefixed by a timestamp) both for installation and upgrade. In the second case, rails will remember the last applied change and handle the rest.